shopfloor/services/location_content_transfer.py

```diff
@@ -913,15 +921,21 @@ def stock_out_line(self, location_id, move_line_id):
         # We need to set qty_done at 0 because otherwise
         # the move_line will not be deleted
         move_line.qty_done = 0
-        move._do_unreserve()
-        move._recompute_state()
+        if self.is_allow_move_create() and self.env.user == move.picking_id.create_uid:
+            # Owned by the user deleting the move
+            move._do_unreserve()
+            move._recompute_state()
+            move._action_cancel()
```

Comment:
Looks good to call `_action_cancel` before before creating the stock issue.
`_action_cancel` is already doing the `_do_unreserve`
```suggestion
            move._action_cancel()
```

Also update docstring